janitor/functions.py

```diff
@@ -3657,6 +3809,63 @@ def to_datetime(
     return df
 
 
+@pf.register_dataframe_method
+def trunc_datetime(datepart: str, timestamp: dt.datetime):
+    """
+    Truncate times down to a user-specified precision of
+    year, month, day, hour, minute, or second.
+
+    Call on datetime object to truncate it.
+    Calling on existing df will not alter the contents
+    of said df.
+
+    Note: Truncating down to a Month or Day will yields 0s,
+    as there is no 0 month or 0 day in most datetime systems.
+
+    :param datepart: Truncation precision, Year, Month, Day,
+        Hour, Minute, Second.
+    :param timestamp: expecting a datetime from python datetime class (dt)
+
+    :returns: a truncated datetime object to
+        the precision specified by datepart.
+    """
+    recurrence = [0, 1, 1, 0, 0, 0]  # [Year, Month, Day, Hour, Minute, Second]
+    datepart = datepart.upper()
+    ENUM = {
+        "YEAR": 0,
+        "MONTH": 1,
+        "DAY": 2,
+        "HOUR": 3,
+        "MINUTE:": 4,
+        "SECOND": 5,
+        0: timestamp.year,
+        1: timestamp.month,
+        2: timestamp.day,
+        3: timestamp.hour,
+        4: timestamp.minute,
+        5: timestamp.second,
+    }
+    try:
+        ENUM[datepart]
+    except KeyError:
+        print("Invalid truncation")
+        print("Please enter Year, Month, Day, Hour, Minute, or Second")
+        return
```

Comment:
For this block, I think the Pythonic pattern is to raise the error loudly and interrupt the program rather than letting it pass through. Here's a suggested modification that would bring this in-line:

```suggestion
    try:
        ENUM[datepart]
    # Capture the error but replace it with explicit instructions.
    except KeyError:
        msg = "Invalid truncation. Please enter any one of 'year', 'month', 'day', 'hour', 'minute' or 'second'."
        raise KeyError(msg)
```

Reply:
**Action Item:** Please change this block according to the suggestion above.